src/command/utils.rs

```diff
@@ -8,8 +8,34 @@ use walkdir::WalkDir;
 
 /// If an explicit path is given, then use it, otherwise assume the current
 /// directory is the crate path.
-pub fn set_crate_path(path: Option<PathBuf>) -> Result<PathBuf, failure::Error> {
-    Ok(path.unwrap_or_else(|| PathBuf::from(".")))
+pub fn get_crate_path(path: Option<PathBuf>) -> Result<PathBuf, failure::Error> {
+    match path {
+        Some(p) => Ok(p),
+        None => find_manifest_from_cwd(),
+    }
+}
+
+/// Search up the path for the manifest file from the current working directory
+/// If we don't find the manifest file then return back the current working directory
+/// to provide the appropriate error
+fn find_manifest_from_cwd() -> Result<PathBuf, failure::Error> {
+    let cwd = PathBuf::from(".");
+    let parent_path = cwd.canonicalize()?;
+    let mut parent_path = parent_path.as_path();
+    let mut manifest_path = parent_path.join("Cargo.toml");
+    loop {
+        if !manifest_path.is_file() {
+            manifest_path = match parent_path.parent() {
+                Some(p) => {
+                    parent_path = p;
+                    p.join("Cargo.toml")
+                }
+                None => break Ok(cwd),
+            };
+        } else {
+            break Ok(parent_path.to_path_buf());
```

Comment:
These `break`s should be `return`.

When it is `break`, I double check if there is a `let foo = loop { ... };` but there isn't, and then I realize it is the last expression in the function, so it is equivalent to `return`... and we can avoid readers doing that same little bit of confusion+understanding in the future by using `return`.

Reply:
I agree - this was my attempt to do things the 'idiomatic' way - it seems from what I've been reading rustwise the preferred thing is to not have explicit return calls. I personally prefer explicit use of return because it makes it clear that is the intention (although some of that is that half the time I automatically add the semi colon and wonder why the compiler is saying I forgot to return a value from the function)